src/lib/exponent/index.ts

```diff
@@ -0,0 +1,103 @@
+import {z} from 'zod';
+
+import {ERRORS} from '../../util/errors';
+import {buildErrorMessage} from '../../util/helpers';
+import {validate} from '../../util/validations';
+
+import {PluginInterface} from '../../interfaces';
+import {PluginParams} from '../../types/common';
+import {ExponentConfig} from './types';
+
+const {InputValidationError} = ERRORS;
+
+export const Exponent = (globalConfig: ExponentConfig): PluginInterface => {
+  const errorBuilder = buildErrorMessage(Exponent.name);
+  const metadata = {
+    kind: 'execute',
+  };
+
+  /**
+   * Checks global config value are valid.
+   */
+  const validateGlobalConfig = () => {
+    const globalConfigSchema = z.object({
+      'input-parameter': z.string().min(1),
+      exponent: z.number().min(1),
+      'output-parameter': z.string().min(1),
+    });
+
+    return validate<z.infer<typeof globalConfigSchema>>(
+      globalConfigSchema,
+      globalConfig
+    );
+  };
+
+  /**
+   * Checks for required fields in input.
+   */
+  const validateSingleInput = (input: PluginParams, inputParameter: string) => {
+    validateParamExists(input, inputParameter);
+    validateNumericString(input[inputParameter]);
+    return input;
+  };
+
+  const validateParamExists = (input: PluginParams, param: string) => {
+    if (!input[param]) {
```

Comment:
If the parameter is `0`, it will throw an error that the parameter is not provided. Change condition to this one `input[param] !== undefined`.